app/bundles/CampaignBundle/Assets/js/campaign.js

```diff
@@ -2418,6 +2423,55 @@ Mautic.previewCampaignLabels = function() {
     });
 }
 
+Mautic.previewCampaignEventDetails = function() {
+    const $allElements = mQuery(Mautic.getCampaignBuilderHtmlElements());
+
+    $allElements.click(function (event) {
+        const $el = mQuery(event.currentTarget);
+        const eventId = $el.data('eventId');
+
+        if (!eventId) return;
+
+        if ($el.hasClass('event-details-visible')) {
+            $el.removeClass('event-details-visible');
+            return;
+        } else {
+            $allElements.removeClass('event-details-visible');
+            $el.addClass('event-details-visible');
+        }
+
+
+        if ($el.data('event-details-loaded')) return;
+
+        mQuery.ajax({
+            url: '/s/campaign/metrics/event-details/' + eventId,
+            success: function (response) {
+                renderEventDetails($el, response);
+            }
```

Comment:
To keep error handling consistent and maintainable, I recommend updating the error callback e.g.:

```suggestion
            },
            error: function (request, textStatus, errorThrown) {
                            Mautic.processAjaxError(request, textStatus, errorThrown);
                        }
```

Reply:
Good catch. I've updated the code to display the error message where the user expects to see the event data.
![image](https://github.com/user-attachments/assets/ea0b8f3c-cfa7-409b-a410-6e7bd53bdae9)